miso-web/src/main/webapp/scripts/freezer_ajax.js

```diff
@@ -309,31 +316,61 @@
       };
       break;
     case 'LOOSE_STORAGE':
-      return function() {
-        var actions = node.item.boxes.map(function(box) {
-          return {
-            name: "View " + box.alias,
-            handler: function() {
-              window.location = window.location.origin + '/miso/box/' + box.id;
-            }
-          };
-        });
-        actions.unshift({
-          name: "Add Box to Storage",
-          handler: assignBox
-        });
-        Utils.showWizardDialog('Boxes in ' + node.item.displayLocation, actions);
-        $('#levelTwoStorageContainer .selected').removeClass('selected');
-        selectedComponent = node.item;
-        node.addClass('selected');
-        $('#editStorageComponentContainer').hide();
-      };
+      return getUnorganizedStorageSelectFunction(node, false, assignBox);
       break;
+    case 'TRAY':
+      return getUnorganizedStorageSelectFunction(node, true, assignBox);
     default:
       throw 'Unexpected box location';
     }
   }
 
+  function getUnorganizedStorageSelectFunction(node, allowEdit, assignBoxFunction) {
+    return function() {
+      var actions = node.item.boxes.map(function(box) {
+        return {
+          name: "View " + box.alias,
+          handler: function() {
+            window.location = window.location.origin + '/miso/box/' + box.id;
+          }
+        };
+      });
+      actions.unshift({
+        name: "Add Box to Storage",
+        handler: assignBoxFunction
+      });
+      Utils.showWizardDialog('Boxes in ' + node.item.displayLocation, actions);
+      $('#levelTwoStorageContainer .selected').removeClass('selected');
+      selectedComponent = node.item;
+      node.addClass('selected');
+      if (allowEdit) {
+        displayEditStorageComponentControls(node.item);
+      } else {
+        $('#editStorageComponentContainer').hide();
+      }
+    };
+  }
+
+  function displayTrayRack(rack) {
+    var table = $('#levelTwoStorageLayout');
+    rack.childLocations.sort(compareLocations).forEach(function(tray) {
+      if (tray.locationUnit != 'TRAY') {
+        throw 'Unexpected location unit';
+      }
+      var row = $('<tr>');
+      var cell = $('<td>');
```

Comment:
Could use `document.createElement('TR'/'TD')` (and getElementById for `table`) to keep it all vanilla JS in this function

Reply:
I'm pretty sure we'll never rip out jQuery. Is there any benefit to avoiding its usage?